File: app/src/main/java/com/zane/smapiinstaller/dto/JsonEditorObject.java
package com.zane.smapiinstaller.dto;

import android.webkit.JavascriptInterface;

import java.util.function.Consumer;

import lombok.AllArgsConstructor;

/**
 * @author Zane
 */
@AllArgsConstructor
public class JsonEditorObject {
    private String text;
    private String mode;
    private String language;
    private boolean editable;
    private int height;
    private Consumer<String> setterCallback;

    @JavascriptInterface
    public String getText() {
        return text;
    }

    @JavascriptInterface
    public void setText(String text) {
        this.text = text;
        if(setterCallback != null) {
            setterCallback.accept(text);
        }
    }

    @JavascriptInterface
    public boolean isEditable() {
        return editable;
    }

    @JavascriptInterface
    public int getHeight() {
        return height;
    }

    @JavascriptInterface
    public String getMode() {
        return mode;
    }

    @JavascriptInterface
    public String getLanguage() {
        return language;
    }
}

File: app/src/main/java/com/zane/smapiinstaller/dto/JsonEditorObject.kt
package com.zane.smapiinstaller.dto

import android.webkit.JavascriptInterface

/**
 * @author Zane
 */
class JsonEditorObject
    (
    private var text: String,
    @get:JavascriptInterface val mode: String,
    @get:JavascriptInterface val language: String,
    @get:JavascriptInterface val isEditable: Boolean,
    @get:JavascriptInterface val height: Int,
    private val setterCallback: ((String) -> Unit)? = null
) {

    @JavascriptInterface
    fun getText(): String {
        return text
    }

    @JavascriptInterface
    fun setText(text: String) {
        this.text = text
        setterCallback?.invoke(text)
    }
}